pkg/controller/nodelink/nodelink_controller.go

```diff
@@ -117,28 +117,32 @@ func indexMachineByInternalIP(object runtime.Object) []string {
 // newReconciler returns a new reconcile.Reconciler
 func newReconciler(mgr manager.Manager) (*ReconcileNodeLink, error) {
 	// set convenient indexers
-	if err := mgr.GetCache().IndexField(&corev1.Node{},
+	if err := mgr.GetCache().IndexField(context.Background(),
```

Comment:
Similarly, the ones in this file should probably be TODO, I imagine we might plumb a context through at some point